blocks/columns/columns.js

```diff
@@ -1,18 +1,32 @@
 export default function decorate(block) {
   const cols = [...block.firstElementChild.children];
   block.classList.add(`columns-${cols.length}-cols`);
+  let downloadableColumns = false;
 
   // setup image columns
   [...block.children].forEach((row) => {
     [...row.children].forEach((col) => {
       const pic = col.querySelector('picture');
+      const link = col.querySelector('a');
+
+      if (pic && link && link.href
+        && new URL(link.href).pathname === new URL(link.innerText).pathname) {
+        link.innerHTML = '';
+        link.appendChild(pic.cloneNode(true));
+        pic.remove();
+        downloadableColumns = true;
+      }
+
       if (pic) {
         const picWrapper = pic.closest('div');
         if (picWrapper && picWrapper.children.length === 1) {
           // picture is only content in column
           picWrapper.classList.add('columns-img-col');
         }
       }
+      if (downloadableColumns) {
+        block.classList.add('centered-text');
```

Comment:
I'm always having a hard time balancing between freedom for the authors and making things too complex. From a gut feeling I would have said the class `centered-text` could also just be something that authors add themself instead of applying it automatically. @dkuntze what is your opinion on this?

Reply:
I don't like the idea of authors having to know class names